Base/Python/slicer/util.py

```diff
@@ -2520,22 +2546,46 @@ def updateTableFromArray(tableNode, narrays, columnNames=None):
 
     if tableNode is None:
         tableNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLTableNode")
-    if isinstance(narrays, np.ndarray) and len(narrays.shape) == 1:
+
+    if isinstance(narrays, np.ndarray) and narrays.dtype.names:
+        # Structured array with named fields
+        fieldNames = narrays.dtype.names
+        if columnNames is None:
+            columnNames = list(fieldNames)
+        ncolumns = [narrays[fieldName] for fieldName in fieldNames]
+    elif isinstance(narrays, np.ndarray) and len(narrays.shape) == 1:
+        # Single 1D array
         ncolumns = [narrays]
     elif isinstance(narrays, np.ndarray) and len(narrays.shape) == 2:
+        # 2D array -> transpose so columns become separate arrays
         ncolumns = narrays.T
     elif isinstance(narrays, tuple) or isinstance(narrays, list):
+        # List or tuple of 1D arrays
         ncolumns = narrays
     else:
-        raise ValueError("Expected narrays is a numpy ndarray, or tuple or list of numpy ndarrays, got %s instead." % (str(type(narrays))))
+        # Unrecognized format
+        msg = (
+            "Expected a structured NumPy array (with named fields), "
+            "a 1D/2D NumPy array, or a list/tuple of 1D arrays; got {} instead."
+        )
+        raise ValueError(msg.format(type(narrays)))
+
     tableNode.RemoveAllColumns()
     # Convert single string to a single-element string list
     if columnNames is None:
         columnNames = []
     if isinstance(columnNames, str):
         columnNames = [columnNames]
+
+    # For each extracted column, convert to a VTK array and add to the table.
     for columnIndex, ncolumn in enumerate(ncolumns):
-        vcolumn = vtk.util.numpy_support.numpy_to_vtk(num_array=ncolumn.ravel(), deep=True, array_type=vtk.VTK_FLOAT)
+        if ncolumn.dtype == np.bool_:
+            # Convert boolean arrays to uint to ensure compatibility with VTK plotting infrastructure:
+            # vtkContext2D does not support vtkBitArray, and plotting calls that rely on
+            # vtkArrayDownCast<vtkFloatArray>() would fail with a bit array.
+            ncolumn = ncolumn.astype("uint")
```

Comment:
While bool does not play nice with plotting, it is quite rare you want to plot a bool function. Instead, bool is useful if you want to have checkboxes in your table.

So, either remove this special conversion (make it the user's responsibility to convert the column type if he wants bools as int) or make it optional (e.g., with an optional parameter like `setBoolAsUint`).

Reply:
Good point, I will add support for boolean array in a follow-up pull request.